### PR TITLE
Add group layer support

### PR DESCRIPTION
**Pull request by Anti-Alias**

### CODE REVIEW

**Review by bjorn — DISMISSED** [review]
Looks great to me!

Unfortunately it will conflict a lot with #135, but that'll be between you and @alexdevteam. :-)

**Comment by alexdevteam** [comment]
I'd really appreciate if this gets delayed until #135, otherwise it's going to be 10x more pain to merge it...

**Comment by Anti-Alias** [comment]
@alexdevteam Okie smokie. It can wait.

**Comment by alexdevteam** [comment]
#135 has been merged now; You are free to reimplement the changes now (and I am really sorry about all the changes and modularization)

File: src/map.rs [hunk 1]
```diff
@@ -103,8 +103,8 @@ impl Map {
     }
 
     /// Get an iterator over all the layers in the map in ascending order of their layer index.
-    pub fn layers(&self) -> LayerIter {
-        LayerIter::new(self)
+    pub fn layers(&self) -> MapLayerIter {
```
Comment (Anti-Alias) [inline src/map.rs:106]: Changed name to disambiguate from new GroupLayerIter

File: src/map.rs [hunk 2]
```diff
@@ -114,18 +114,18 @@ impl Map {
 }
 
 /// An iterator that iterates over all the layers in a map, obtained via [`Map::layers`].
-pub struct LayerIter<'map> {
+pub struct MapLayerIter<'map> {
```
Comment (Anti-Alias) [inline src/map.rs:117]: Changed name to disambiguate from new GroupLayerIter

Comment (Anti-Alias) [reply src/map.rs:117]: Hah, no need to apologize.
I must have picked the worst time to do that first PR yesterday!

**Comment by Anti-Alias** [comment]
Hah, no need to apologize.
I must have picked the worst time to do that PR yesterday.

File: tests/lib.rs [hunk 3]
```diff
@@ -196,10 +203,10 @@ fn test_object_group_property() {
     let mut cache = FilesystemResourceCache::new();
 
     let r = Map::parse_file("assets/tiled_object_groups.tmx", &mut cache).unwrap();
-    let prop_value: bool = if let Some(&PropertyValue::BoolValue(ref v)) = r
-        .layers()
-        .nth(1)
-        .unwrap()
+    let group_layer = r.get_layer(1).unwrap();
```
Comment (alexdevteam) [inline tests/lib.rs:206]: Why'd this change? Does the object layer example have group layers?

Comment (bjorn) [reply tests/lib.rs:206]: Yeah, it was using a group all along, but apparently this was just ignored by the XML parser before:

https://github.com/mapeditor/rs-tiled/blob/df98e9f35afb6d6a7d3b431537682ee67382fb68/assets/tiled_object_groups.tmx#L17-L23

**Review by bjorn — APPROVED** [review]

**Review by alexdevteam — APPROVED** [review]

**Pull request merged**